### PR TITLE
Migrating RS.Backup to Swagger

### PR DESCRIPTION
**Pull request by dragonfly91**
## Description

Moved the RecoveryServices.Backup module from underlying Hyak dependency to Swagger.
---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThrough` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Test/packages.config [hunk 1]
```diff
@@ -5,7 +5,8 @@
   <package id="Microsoft.Azure.Common.Dependencies" version="1.0.0" targetFramework="net45" />
   <package id="Microsoft.Azure.Management.Resources" version="2.18.14-preview" targetFramework="net45" />
   <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="2.28.1" targetFramework="net45" />
-  <package id="Microsoft.Rest.ClientRuntime" version="2.3.2" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.3" targetFramework="net45" />
```
Comment (markcowl) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Test/packages.config]: The versions of Microsoft.Rest.ClientRuntime and Microsoft.Rest.ClientRuntime.Azure need to be the same across all modules.  Please standardize on the version currently in use.

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup/packages.config [hunk 2]
```diff
@@ -2,13 +2,14 @@
 <packages>
   <package id="Hyak.Common" version="1.0.3" targetFramework="net45" />
   <package id="Microsoft.Azure.Common" version="2.1.0" targetFramework="net45" />
-  <package id="Microsoft.Rest.ClientRuntime" version="2.3.2" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.3" targetFramework="net45" />
```
Comment (markcowl) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup/packages.config]: see comment on runtime assemblies above

**Comment by dragonfly91** [comment]
@markcowl,

We have a dependency on the 2.3.3 version of the Microsoft.Rest.ClientRuntime.dll. It is not possible for us to use the 2.3.2 version. The review comment was to normalize the versions across all the modules. The only viable option that I can see here is to upgrade all the modules in PS to use 2.3.3 version of the dll. 

Can you suggest what is the expectation here?

1.	Should we upgrade all the modules to 2.3.3?
a.	How to ensure there is no regression? Will running checkin-tests, as described in the PS wiki, be enough?
b.	Should this change be sent as a separate PR?
2.	Or will you be able to upgrade the modules? I’ve come across a case when you had to do this to unblock the KeyVault module (https://github.com/Azure/azure-powershell/pull/2616)

File: ChangeLog.md [hunk 3]
```diff
@@ -98,6 +98,8 @@
     * Add-AzureRmAccount
         - Add position for Credential parameter so the following command is allowed: Add-AzureRmAccount (Get-Credential)
         - Updated parameter sets so the SubscriptionId and SubscriptionName are mutually exclusive
+* RecoveryServices.Backup
```
Comment (shahabhijeet) [inline ChangeLog.md]: You do not change this file. Rather add the changes in your respective directory. you should have a changelog.md in your RP root folder
Follow the guidelines and update changelog.md
[https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log)

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup/packages.config (hunk 2 above)
Comment (markcowl) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup/packages.config]: actually, we will make the change for other libraries.  You cna ignore this

**Comment by markcowl** [comment]
@dragonfly91 Please make the change Abhijeet suggests for changelog (move the changes into the changelog for RS.Backup, rather than the central changelog)  and we will handle updating the runtime versions for other libraries.

**Comment by dragonfly91** [comment]
@markcowl we've made the changelog fix. Can you please merge?

**Comment by markcowl** [comment]
@dragonfly91 There are merge conflicts, please pull the latest.

**Comment by dragonfly91** [comment]
@markcowl I've resolved the merge conflicts. However, it doesn't seem that the projects have been moved to ClientRuntime 2.3.3. Is this yet to be done? In order for our test cases to run, I have kept our projects on 2.3.3.

**Comment by markcowl** [comment]
on demand run here:http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1312/ 

LGTM once the on-demand run passes

**Pull request merged**